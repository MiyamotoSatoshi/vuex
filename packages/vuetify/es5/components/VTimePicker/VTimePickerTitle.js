'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

require('../../../src/stylus/components/_time-picker-title.styl');

var _pickerButton = require('../../mixins/picker-button');

var _pickerButton2 = _interopRequireDefault(_pickerButton);

var _util = require('../VDatePicker/util');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* @vue/component */

// Mixins
exports.default = {
    name: 'v-time-picker-title',
    mixins: [_pickerButton2.default],
    props: {
        ampm: Boolean,
        hour: Number,
        minute: Number,
        period: {
            type: String,
            validator: function validator(period) {
                return period === 'am' || period === 'pm';
            }
        },
        readonly: Boolean,
        selectingHour: Boolean
    },
    methods: {
        genTime: function genTime() {
            var hour = this.hour;
            if (this.ampm) {
                hour = hour ? (hour - 1) % 12 + 1 : 12;
            }
            var displayedHour = this.hour == null ? '--' : this.ampm ? hour : (0, _util.pad)(hour);
            var displayedMinute = this.minute == null ? '--' : (0, _util.pad)(this.minute);
            return this.$createElement('div', {
                'class': 'v-time-picker-title__time'
            }, [this.genPickerButton('selectingHour', true, displayedHour), this.$createElement('span', ':'), this.genPickerButton('selectingHour', false, displayedMinute)]);
        },
        genAmPm: function genAmPm() {
            return this.$createElement('div', {
                staticClass: 'v-time-picker-title__ampm'
            }, [this.genPickerButton('period', 'am', 'am', this.readonly), this.genPickerButton('period', 'pm', 'pm', this.readonly)]);
        }
    },
    render: function render(h) {
        return h('div', {
            staticClass: 'v-time-picker-title'
        }, [this.genTime(), this.ampm ? this.genAmPm() : null]);
    }
};
// Utils
//# sourceMappingURL=VTimePickerTitle.js.map