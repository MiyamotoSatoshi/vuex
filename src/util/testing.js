import Vue from 'vue'
import load from '~util/load'
import { mount, shallow } from 'avoriaz'

export function test(name, cb) {
  Vue.prototype.$vuetify = { load }

  beforeEach(() => {
    rafPolyfill(window)
  })

  describe(name, () => cb({
    functionalContext,
    mount,
    shallow
  }))
}

export function functionalContext(context = {}) {
  return {
    context: Object.assign({
      data: {},
      props: {}
    }, context)
  }
}


//requestAnimationFrame polyfill | Milos Djakonovic ( @Miloshio ) | MIT | https://github.com/milosdjakonovic/requestAnimationFrame-polyfill
export function rafPolyfill(w) {
  /**
   *
   * How many times should polyfill call
   * update callback? By canon, it should
   * be 60 times per second, so that ideal
   * framerate 60fps could be reached.
   *
   * However, even native implementations
   * of requestAnimationFrame often cannot
   * do 60fps, but, unlike any polyfill,
   * they can synchronise achievable fps
   * rate with screen refresh rate.
   *
   * So, leave this value 1000/60 unless
   * you target specific browser on spec
   * ific device that is going to work
   * better with custom value. I think
   * that this is the longest comment I've
   * written on single variable so far.
  **/
  var FRAME_RATE_INTERVAL = 1000/60,

  /**
   * All queued callbacks in given cycle.
  **/
  allCallbacks = [],

  executeAllScheduled = false,

  shouldCheckCancelRaf = false,

  /**
   * Callbacks queued for cancellation.
  **/
  callbacksForCancellation = [],

  /**
   * Should callback be cancelled?
   * @param cb - callback
  **/
  isToBeCancelled = function(cb){
    for(var i=0;i<callbacksForCancellation.length;i++){
      if(callbacksForCancellation[i] === cb ){
        callbacksForCancellation.splice(i,1);
        return true;
      }
    }
  },



  /**
   *
   * Executes all (surprise) callbacks in
   * and removes them from callback queue.
   *
  **/
  executeAll = function(){
    executeAllScheduled = false;
    var _allCallbacks = allCallbacks;
    allCallbacks = [];
    for(var i=0;i<_allCallbacks.length;i++){
      if(shouldCheckCancelRaf===true){
        if (isToBeCancelled(_allCallbacks[i])){
          shouldCheckCancelRaf = false;
          return;
        }
      }
      _allCallbacks[i].apply(w, [ new Date().getTime() ] );
    }
  },

  /**
   *
   * requestAnimationFrame polyfill
   * @param callback - callback to be queued & executed | executed
   * @return callback
   *
  **/
  raf = function(callback){
    allCallbacks.push(callback);
    if(executeAllScheduled===false){
      w.setTimeout(executeAll, FRAME_RATE_INTERVAL);
      executeAllScheduled = true;
    }
    return callback;
  },

  /**
   *
   * Cancels raf.
  **/
  cancelRaf = function(callback){
    callbacksForCancellation.push(callback);
    shouldCheckCancelRaf = true;
  },


  //https://gist.github.com/paulirish/1579671
  vendors = ['ms', 'moz', 'webkit', 'o'];

    for(var x = 0; x < vendors.length && !w.requestAnimationFrame; ++x) {
        w.requestAnimationFrame = w[vendors[x]+'RequestAnimationFrame'];
        w.cancelAnimationFrame = w[vendors[x]+'CancelAnimationFrame']
        || w[vendors[x]+'CancelRequestAnimationFrame'];
    }

  if (!w.requestAnimationFrame) w.requestAnimationFrame = raf;
  if (!w.cancelAnimationFrame)  w.cancelAnimationFrame  = cancelRaf;

}
