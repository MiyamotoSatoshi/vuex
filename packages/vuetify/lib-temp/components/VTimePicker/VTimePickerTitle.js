import '../../stylus/components/_time-picker-title.styl';
// Mixins
import PickerButton from '../../mixins/picker-button';
// Utils
import { pad } from '../VDatePicker/util';
/* @vue/component */
export default {
    name: 'v-time-picker-title',
    mixins: [PickerButton],
    props: {
        ampm: Boolean,
        hour: Number,
        minute: Number,
        period: {
            type: String,
            validator: period => period === 'am' || period === 'pm'
        },
        readonly: Boolean,
        selectingHour: Boolean
    },
    methods: {
        genTime() {
            let hour = this.hour;
            if (this.ampm) {
                hour = hour ? ((hour - 1) % 12 + 1) : 12;
            }
            const displayedHour = this.hour == null ? '--' : this.ampm ? hour : pad(hour);
            const displayedMinute = this.minute == null ? '--' : pad(this.minute);
            return this.$createElement('div', {
                'class': 'v-time-picker-title__time'
            }, [
                this.genPickerButton('selectingHour', true, displayedHour),
                this.$createElement('span', ':'),
                this.genPickerButton('selectingHour', false, displayedMinute)
            ]);
        },
        genAmPm() {
            return this.$createElement('div', {
                staticClass: 'v-time-picker-title__ampm'
            }, [
                this.genPickerButton('period', 'am', 'am', this.readonly),
                this.genPickerButton('period', 'pm', 'pm', this.readonly)
            ]);
        }
    },
    render(h) {
        return h('div', {
            staticClass: 'v-time-picker-title'
        }, [
            this.genTime(),
            this.ampm ? this.genAmPm() : null
        ]);
    }
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiVlRpbWVQaWNrZXJUaXRsZS5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIi4uLy4uLy4uL3NyYy9jb21wb25lbnRzL1ZUaW1lUGlja2VyL1ZUaW1lUGlja2VyVGl0bGUuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUEsT0FBTyxpREFBaUQsQ0FBQTtBQUV4RCxTQUFTO0FBQ1QsT0FBTyxZQUFZLE1BQU0sNEJBQTRCLENBQUE7QUFFckQsUUFBUTtBQUNSLE9BQU8sRUFBRSxHQUFHLEVBQUUsTUFBTSxxQkFBcUIsQ0FBQTtBQUV6QyxvQkFBb0I7QUFDcEIsZUFBZTtJQUNiLElBQUksRUFBRSxxQkFBcUI7SUFFM0IsTUFBTSxFQUFFLENBQUMsWUFBWSxDQUFDO0lBRXRCLEtBQUssRUFBRTtRQUNMLElBQUksRUFBRSxPQUFPO1FBQ2IsSUFBSSxFQUFFLE1BQU07UUFDWixNQUFNLEVBQUUsTUFBTTtRQUNkLE1BQU0sRUFBRTtZQUNOLElBQUksRUFBRSxNQUFNO1lBQ1osU0FBUyxFQUFFLE1BQU0sQ0FBQyxFQUFFLENBQUMsTUFBTSxLQUFLLElBQUksSUFBSSxNQUFNLEtBQUssSUFBSTtTQUN4RDtRQUNELFFBQVEsRUFBRSxPQUFPO1FBQ2pCLGFBQWEsRUFBRSxPQUFPO0tBQ3ZCO0lBRUQsT0FBTyxFQUFFO1FBQ1AsT0FBTztZQUNMLElBQUksSUFBSSxHQUFHLElBQUksQ0FBQyxJQUFJLENBQUE7WUFDcEIsSUFBSSxJQUFJLENBQUMsSUFBSSxFQUFFO2dCQUNiLElBQUksR0FBRyxJQUFJLENBQUMsQ0FBQyxDQUFDLENBQUMsQ0FBQyxJQUFJLEdBQUcsQ0FBQyxDQUFDLEdBQUcsRUFBRSxHQUFHLENBQUMsQ0FBQyxDQUFDLENBQUMsQ0FBQyxFQUFFLENBQUE7YUFDekM7WUFFRCxNQUFNLGFBQWEsR0FBRyxJQUFJLENBQUMsSUFBSSxJQUFJLElBQUksQ0FBQyxDQUFDLENBQUMsSUFBSSxDQUFDLENBQUMsQ0FBQyxJQUFJLENBQUMsSUFBSSxDQUFDLENBQUMsQ0FBQyxJQUFJLENBQUMsQ0FBQyxDQUFDLEdBQUcsQ0FBQyxJQUFJLENBQUMsQ0FBQTtZQUM3RSxNQUFNLGVBQWUsR0FBRyxJQUFJLENBQUMsTUFBTSxJQUFJLElBQUksQ0FBQyxDQUFDLENBQUMsSUFBSSxDQUFDLENBQUMsQ0FBQyxHQUFHLENBQUMsSUFBSSxDQUFDLE1BQU0sQ0FBQyxDQUFBO1lBRXJFLE9BQU8sSUFBSSxDQUFDLGNBQWMsQ0FBQyxLQUFLLEVBQUU7Z0JBQ2hDLE9BQU8sRUFBRSwyQkFBMkI7YUFDckMsRUFBRTtnQkFDRCxJQUFJLENBQUMsZUFBZSxDQUFDLGVBQWUsRUFBRSxJQUFJLEVBQUUsYUFBYSxDQUFDO2dCQUMxRCxJQUFJLENBQUMsY0FBYyxDQUFDLE1BQU0sRUFBRSxHQUFHLENBQUM7Z0JBQ2hDLElBQUksQ0FBQyxlQUFlLENBQUMsZUFBZSxFQUFFLEtBQUssRUFBRSxlQUFlLENBQUM7YUFDOUQsQ0FBQyxDQUFBO1FBQ0osQ0FBQztRQUNELE9BQU87WUFDTCxPQUFPLElBQUksQ0FBQyxjQUFjLENBQUMsS0FBSyxFQUFFO2dCQUNoQyxXQUFXLEVBQUUsMkJBQTJCO2FBQ3pDLEVBQUU7Z0JBQ0QsSUFBSSxDQUFDLGVBQWUsQ0FBQyxRQUFRLEVBQUUsSUFBSSxFQUFFLElBQUksRUFBRSxJQUFJLENBQUMsUUFBUSxDQUFDO2dCQUN6RCxJQUFJLENBQUMsZUFBZSxDQUFDLFFBQVEsRUFBRSxJQUFJLEVBQUUsSUFBSSxFQUFFLElBQUksQ0FBQyxRQUFRLENBQUM7YUFDMUQsQ0FBQyxDQUFBO1FBQ0osQ0FBQztLQUNGO0lBRUQsTUFBTSxDQUFFLENBQUM7UUFDUCxPQUFPLENBQUMsQ0FBQyxLQUFLLEVBQUU7WUFDZCxXQUFXLEVBQUUscUJBQXFCO1NBQ25DLEVBQUU7WUFDRCxJQUFJLENBQUMsT0FBTyxFQUFFO1lBQ2QsSUFBSSxDQUFDLElBQUksQ0FBQyxDQUFDLENBQUMsSUFBSSxDQUFDLE9BQU8sRUFBRSxDQUFDLENBQUMsQ0FBQyxJQUFJO1NBQ2xDLENBQUMsQ0FBQTtJQUNKLENBQUM7Q0FDRixDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0ICcuLi8uLi9zdHlsdXMvY29tcG9uZW50cy9fdGltZS1waWNrZXItdGl0bGUuc3R5bCdcblxuLy8gTWl4aW5zXG5pbXBvcnQgUGlja2VyQnV0dG9uIGZyb20gJy4uLy4uL21peGlucy9waWNrZXItYnV0dG9uJ1xuXG4vLyBVdGlsc1xuaW1wb3J0IHsgcGFkIH0gZnJvbSAnLi4vVkRhdGVQaWNrZXIvdXRpbCdcblxuLyogQHZ1ZS9jb21wb25lbnQgKi9cbmV4cG9ydCBkZWZhdWx0IHtcbiAgbmFtZTogJ3YtdGltZS1waWNrZXItdGl0bGUnLFxuXG4gIG1peGluczogW1BpY2tlckJ1dHRvbl0sXG5cbiAgcHJvcHM6IHtcbiAgICBhbXBtOiBCb29sZWFuLFxuICAgIGhvdXI6IE51bWJlcixcbiAgICBtaW51dGU6IE51bWJlcixcbiAgICBwZXJpb2Q6IHtcbiAgICAgIHR5cGU6IFN0cmluZyxcbiAgICAgIHZhbGlkYXRvcjogcGVyaW9kID0+IHBlcmlvZCA9PT0gJ2FtJyB8fCBwZXJpb2QgPT09ICdwbSdcbiAgICB9LFxuICAgIHJlYWRvbmx5OiBCb29sZWFuLFxuICAgIHNlbGVjdGluZ0hvdXI6IEJvb2xlYW5cbiAgfSxcblxuICBtZXRob2RzOiB7XG4gICAgZ2VuVGltZSAoKSB7XG4gICAgICBsZXQgaG91ciA9IHRoaXMuaG91clxuICAgICAgaWYgKHRoaXMuYW1wbSkge1xuICAgICAgICBob3VyID0gaG91ciA/ICgoaG91ciAtIDEpICUgMTIgKyAxKSA6IDEyXG4gICAgICB9XG5cbiAgICAgIGNvbnN0IGRpc3BsYXllZEhvdXIgPSB0aGlzLmhvdXIgPT0gbnVsbCA/ICctLScgOiB0aGlzLmFtcG0gPyBob3VyIDogcGFkKGhvdXIpXG4gICAgICBjb25zdCBkaXNwbGF5ZWRNaW51dGUgPSB0aGlzLm1pbnV0ZSA9PSBudWxsID8gJy0tJyA6IHBhZCh0aGlzLm1pbnV0ZSlcblxuICAgICAgcmV0dXJuIHRoaXMuJGNyZWF0ZUVsZW1lbnQoJ2RpdicsIHtcbiAgICAgICAgJ2NsYXNzJzogJ3YtdGltZS1waWNrZXItdGl0bGVfX3RpbWUnXG4gICAgICB9LCBbXG4gICAgICAgIHRoaXMuZ2VuUGlja2VyQnV0dG9uKCdzZWxlY3RpbmdIb3VyJywgdHJ1ZSwgZGlzcGxheWVkSG91ciksXG4gICAgICAgIHRoaXMuJGNyZWF0ZUVsZW1lbnQoJ3NwYW4nLCAnOicpLFxuICAgICAgICB0aGlzLmdlblBpY2tlckJ1dHRvbignc2VsZWN0aW5nSG91cicsIGZhbHNlLCBkaXNwbGF5ZWRNaW51dGUpXG4gICAgICBdKVxuICAgIH0sXG4gICAgZ2VuQW1QbSAoKSB7XG4gICAgICByZXR1cm4gdGhpcy4kY3JlYXRlRWxlbWVudCgnZGl2Jywge1xuICAgICAgICBzdGF0aWNDbGFzczogJ3YtdGltZS1waWNrZXItdGl0bGVfX2FtcG0nXG4gICAgICB9LCBbXG4gICAgICAgIHRoaXMuZ2VuUGlja2VyQnV0dG9uKCdwZXJpb2QnLCAnYW0nLCAnYW0nLCB0aGlzLnJlYWRvbmx5KSxcbiAgICAgICAgdGhpcy5nZW5QaWNrZXJCdXR0b24oJ3BlcmlvZCcsICdwbScsICdwbScsIHRoaXMucmVhZG9ubHkpXG4gICAgICBdKVxuICAgIH1cbiAgfSxcblxuICByZW5kZXIgKGgpIHtcbiAgICByZXR1cm4gaCgnZGl2Jywge1xuICAgICAgc3RhdGljQ2xhc3M6ICd2LXRpbWUtcGlja2VyLXRpdGxlJ1xuICAgIH0sIFtcbiAgICAgIHRoaXMuZ2VuVGltZSgpLFxuICAgICAgdGhpcy5hbXBtID8gdGhpcy5nZW5BbVBtKCkgOiBudWxsXG4gICAgXSlcbiAgfVxufVxuIl19