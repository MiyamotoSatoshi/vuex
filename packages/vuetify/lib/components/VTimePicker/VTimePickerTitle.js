import '../../../src/stylus/components/_time-picker-title.styl';
// Mixins
import PickerButton from '../../mixins/picker-button';
// Utils
import { pad } from '../VDatePicker/util';
/* @vue/component */
export default {
    name: 'v-time-picker-title',
    mixins: [PickerButton],
    props: {
        ampm: Boolean,
        hour: Number,
        minute: Number,
        period: {
            type: String,
            validator: function validator(period) {
                return period === 'am' || period === 'pm';
            }
        },
        readonly: Boolean,
        selectingHour: Boolean
    },
    methods: {
        genTime: function genTime() {
            var hour = this.hour;
            if (this.ampm) {
                hour = hour ? (hour - 1) % 12 + 1 : 12;
            }
            var displayedHour = this.hour == null ? '--' : this.ampm ? hour : pad(hour);
            var displayedMinute = this.minute == null ? '--' : pad(this.minute);
            return this.$createElement('div', {
                'class': 'v-time-picker-title__time'
            }, [this.genPickerButton('selectingHour', true, displayedHour), this.$createElement('span', ':'), this.genPickerButton('selectingHour', false, displayedMinute)]);
        },
        genAmPm: function genAmPm() {
            return this.$createElement('div', {
                staticClass: 'v-time-picker-title__ampm'
            }, [this.genPickerButton('period', 'am', 'am', this.readonly), this.genPickerButton('period', 'pm', 'pm', this.readonly)]);
        }
    },
    render: function render(h) {
        return h('div', {
            staticClass: 'v-time-picker-title'
        }, [this.genTime(), this.ampm ? this.genAmPm() : null]);
    }
};
//# sourceMappingURL=VTimePickerTitle.js.map