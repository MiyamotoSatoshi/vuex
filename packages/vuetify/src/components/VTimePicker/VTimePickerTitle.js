import '../../stylus/components/_time-picker-title.styl'

// Mixins
import PickerButton from '../../mixins/picker-button'

// Utils
import { pad } from '../VDatePicker/util'

import { selectingTimes } from './VTimePicker'

/* @vue/component */
export default {
  name: 'v-time-picker-title',

  mixins: [PickerButton],

  props: {
    ampm: Boolean,
    disabled: Boolean,
    hour: Number,
    minute: Number,
    second: Number,
    period: {
      type: String,
      validator: period => period === 'am' || period === 'pm'
    },
    readonly: Boolean,
    useSeconds: Boolean,
    selecting: Number
  },

  methods: {
    genTime () {
      let hour = this.hour
      if (this.ampm) {
        hour = hour ? ((hour - 1) % 12 + 1) : 12
      }

      const displayedHour = this.hour == null ? '--' : this.ampm ? hour : pad(hour)
      const displayedMinute = this.minute == null ? '--' : pad(this.minute)
      const displayedSecond = this.second == null ? '--' : pad(this.second)

      const titleContent = [
        this.genPickerButton('selecting', selectingTimes.hour, displayedHour, this.disabled),
        this.$createElement('span', ':'),
        this.genPickerButton('selecting', selectingTimes.minute, displayedMinute, this.disabled)
      ]
      if (this.useSeconds) {
        titleContent.push(this.$createElement('span', ':'))
        titleContent.push(this.genPickerButton('selecting', selectingTimes.second, displayedSecond, this.disabled))
      }
      return this.$createElement('div', {
        'class': 'v-time-picker-title__time'
      }, titleContent)
    },
    genAmPm () {
      return this.$createElement('div', {
        staticClass: 'v-time-picker-title__ampm'
      }, [
        this.genPickerButton('period', 'am', 'am', this.disabled || this.readonly),
        this.genPickerButton('period', 'pm', 'pm', this.disabled || this.readonly)
      ])
    }
  },

  render (h) {
    return h('div', {
      staticClass: 'v-time-picker-title'
    }, [
      this.genTime(),
      this.ampm ? this.genAmPm() : null
    ])
  }
}
