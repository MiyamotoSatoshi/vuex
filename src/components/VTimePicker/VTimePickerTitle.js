import '../../stylus/components/_time-picker-title.styl'

// Mixins
import PickerButton from '../../mixins/picker-button'

// Utils
import { pad } from '../VDatePicker/util'

/* @vue/component */
export default {
  name: 'v-time-picker-title',

  mixins: [PickerButton],

  props: {
    ampm: Boolean,
    hour: Number,
    minute: Number,
    period: {
      type: String,
      validator: period => period === 'am' || period === 'pm'
    },
    selectingHour: Boolean
  },

  methods: {
    genTime () {
      let hour = this.hour
      if (this.ampm) {
        hour = hour ? ((hour - 1) % 12 + 1) : 12
      }

      const displayedHour = this.hour == null ? '--' : this.ampm ? hour : pad(hour)
      const displayedMinute = this.minute == null ? '--' : pad(this.minute)

      return this.$createElement('div', {
        'class': 'v-time-picker-title__time'
      }, [
        this.genPickerButton('selectingHour', true, displayedHour),
        this.$createElement('span', ':'),
        this.genPickerButton('selectingHour', false, displayedMinute)
      ])
    },
    genAmPm () {
      return this.$createElement('div', {
        staticClass: 'v-time-picker-title__ampm'
      }, [
        this.genPickerButton('period', 'am', 'am'),
        this.genPickerButton('period', 'pm', 'pm')
      ])
    }
  },

  render (h) {
    return h('div', {
      staticClass: 'v-time-picker-title'
    }, [
      this.genTime(),
      this.ampm ? this.genAmPm() : null
    ])
  }
}
