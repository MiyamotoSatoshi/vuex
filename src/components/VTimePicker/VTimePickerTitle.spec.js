import VTimePickerTitle from './VTimePickerTitle'
import { test } from '@util/testing'

test('VTimePickerTitle.js', ({ mount }) => {
  it('should render component in 24hr', () => {
    const wrapper = mount(VTimePickerTitle, {
      propsData: {
        value: '14:13',
        ampm: false
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component in 12hr', () => {
    const wrapper = mount(VTimePickerTitle, {
      propsData: {
        value: '14:13',
        ampm: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component when selecting hour', () => {
    const wrapper = mount(VTimePickerTitle, {
      propsData: {
        value: '14:13',
        selectingHour: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit event when clicked on am/pm', async () => {
    const wrapper = mount(VTimePickerTitle, {
      propsData: {
        value: '14:13',
        ampm: true
      }
    })

    const period = jest.fn()
    wrapper.vm.$on('update:period', period)

    wrapper.find('.time-picker-title__ampm .picker__title__btn.active')[0].trigger('click')
    expect(period).not.toBeCalled()
    wrapper.find('.time-picker-title__ampm .picker__title__btn:not(.active)')[0].trigger('click')
    expect(period).toBeCalledWith('am')

    wrapper.setProps({
      value: '2:13'
    })
    wrapper.find('.time-picker-title__ampm .picker__title__btn:not(.active)')[0].trigger('click')
    expect(period).toBeCalledWith('pm')
  })

  it('should emit event when clicked on hours/minutes', async () => {
    const wrapper = mount(VTimePickerTitle, {
      propsData: {
        value: '14:13'
      }
    })

    const selectingHour = jest.fn()
    wrapper.vm.$on('update:selectingHour', selectingHour)

    wrapper.find('.time-picker-title__time .picker__title__btn')[1].trigger('click')
    expect(selectingHour).not.toBeCalled()
    wrapper.find('.time-picker-title__time .picker__title__btn')[0].trigger('click')
    expect(selectingHour).toBeCalledWith(true)
    wrapper.setProps({ selectingHour: true })
    await wrapper.vm.$nextTick()
    wrapper.find('.time-picker-title__time .picker__title__btn')[1].trigger('click')
    expect(selectingHour).toBeCalledWith(false)
  })
})
