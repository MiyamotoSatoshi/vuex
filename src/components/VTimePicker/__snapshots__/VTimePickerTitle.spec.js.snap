// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VTimePickerTitle.js should render component in 12hr 1`] = `

<div class="time-picker-title">
  <div class="time-picker-title__time">
    <div class="picker__title__btn">
      2
    </div>
    <span>
      :
    </span>
    <div class="picker__title__btn active">
      13
    </div>
  </div>
  <div class="time-picker-title__ampm">
    <div class="picker__title__btn">
      am
    </div>
    <div class="picker__title__btn active">
      pm
    </div>
  </div>
</div>

`;

exports[`VTimePickerTitle.js should render component in 24hr 1`] = `

<div class="time-picker-title">
  <div class="time-picker-title__time">
    <div class="picker__title__btn">
      14
    </div>
    <span>
      :
    </span>
    <div class="picker__title__btn active">
      13
    </div>
  </div>
</div>

`;

exports[`VTimePickerTitle.js should render component when selecting hour 1`] = `

<div class="time-picker-title">
  <div class="time-picker-title__time">
    <div class="picker__title__btn active">
      14
    </div>
    <span>
      :
    </span>
    <div class="picker__title__btn">
      13
    </div>
  </div>
</div>

`;
